encoding/template/run_length_decode.go

```diff
@@ -0,0 +1,70 @@
+package template
+
+import (
+	"encoding/binary"
+
+	"github.com/xichen2020/eventdb/x/io"
+)
+
+// ValueUnmarshalFn reads a GenericValue from an `io.Reader`.
+type ValueUnmarshalFn func(reader io.Reader) (GenericValue, error)
+
+// runLengthDecodeValue run length decodes a stream of GenericValues.
+func runLengthDecodeValue(
+	reader io.Reader,
+	unmarshalFn ValueUnmarshalFn,
+) *RunLengthValueIterator {
+	return newRunLengthValueIterator(reader, unmarshalFn)
+}
+
+// RunLengthValueIterator iterates over a run length encoded stream of GenericValue data.
+type RunLengthValueIterator struct {
+	reader      io.Reader
+	unmarshalFn ValueUnmarshalFn
```

Comment:
`readValue ReadValueFn` might be a more proper name.

Reply:
👍